spring-integration-core/src/main/java/org/springframework/integration/config/annotation/AbstractMethodAnnotationPostProcessor.java

```diff
@@ -50,6 +51,7 @@
 import org.springframework.integration.channel.DirectChannel;
 import org.springframework.integration.channel.MessagePublishingErrorHandler;
 import org.springframework.integration.config.IntegrationConfigUtils;
+import org.springframework.integration.config.annotation.MessagingAnnotationPostProcessor.Disposables;
```

Comment:
This is going to cause class tangles: the `MessagingAnnotationPostProcessor` uses all the `AbstractMethodAnnotationPostProcessor` implementations.

Maybe just make that `Disposables` as package protected and refer to it everywhere it is necessary?